### PR TITLE
Remove usage of FileSwitchDirectory 

### PR DESCRIPTION
**Pull request by s1monw**
We are still using `FileSwitchDirectory` in the case a user configures file based pre-load of mmaps. This is trappy for multiple reasons if the both directories used by `FileSwitchDirectory` point to the same filesystem directory. One issue is [LUCENE-8835](https://issues.apache.org/jira/browse/LUCENE-8835) that cause issues like #37111 - unless LUCENE-8835 isn't fixed we should not use it in elasticsearch. Instead we use a similar trick as we use for HybridFS and subclass mmap directory directly.

### CODE REVIEW

**Comment by s1monw** [comment]
this becomes even more pressing with https://issues.apache.org/jira/browse/LUCENE-8853

File: server/src/main/java/org/elasticsearch/index/store/FsDirectoryFactory.java [hunk 1]
```diff
@@ -148,26 +158,59 @@ public IndexInput openInput(String name, IOContext context) throws IOException {
...
+                ensureCanRead(name);
+                return delegate.openInput(name, context);
             }
+            return super.openInput(name, context);
         }
 
         @Override
-        public void close() throws IOException {
-            IOUtils.close(super::close, randomAccessDirectory);
+        public synchronized void close() throws IOException {
```
Comment (jimczi) [inline server/src/main/java/org/elasticsearch/index/store/FsDirectoryFactory.java]: why does it need to be synchronized ?

**Review by jimczi — APPROVED** [review]
I left one question, LGTM otherwise

File: server/src/main/java/org/elasticsearch/index/store/FsDirectoryFactory.java (hunk 1 above)
Comment (s1monw) [reply server/src/main/java/org/elasticsearch/index/store/FsDirectoryFactory.java]: that happens automatically since it's synced in the super class

**Pull request merged**